tests/job.go

```diff
@@ -17,30 +17,53 @@ import (
 func WaitForJobToSucceed(virtClient *kubecli.KubevirtClient, job *batchv1.Job, timeoutSec time.Duration) {
 	EventuallyWithOffset(1, func() bool {
 		job, err := (*virtClient).BatchV1().Jobs(job.Namespace).Get(job.Name, metav1.GetOptions{})
-		ExpectWithOffset(1, err).ToNot(HaveOccurred())
-		Expect(job.Status.Failed).NotTo(Equal(*job.Spec.BackoffLimit), "Job was expected to succeed but failed")
-		return job.Status.Succeeded > 0
+		ExpectWithOffset(2, err).ToNot(HaveOccurred())
+		for _, c := range job.Status.Conditions {
+			switch c.Type {
+			case batchv1.JobComplete:
+				if c.Status == k8sv1.ConditionTrue {
+					return true
+				}
+			case batchv1.JobFailed:
+				ExpectWithOffset(2, c.Status).NotTo(Equal(k8sv1.ConditionTrue), "Job should succeed")
+			}
+		}
+		return false
 	}, timeoutSec*time.Second, 1*time.Second).Should(BeTrue(), "Job should succeed")
-
 }
 
 func WaitForJobToFail(virtClient *kubecli.KubevirtClient, job *batchv1.Job, timeoutSec time.Duration) {
-	EventuallyWithOffset(1, func() int32 {
+	EventuallyWithOffset(1, func() bool {
 		job, err := (*virtClient).BatchV1().Jobs(job.Namespace).Get(job.Name, metav1.GetOptions{})
-		ExpectWithOffset(1, err).ToNot(HaveOccurred())
-		Expect(job.Status.Succeeded).ToNot(BeNumerically(">", 0), "Job should not succeed")
-		return job.Status.Failed
-	}, timeoutSec*time.Second, 1*time.Second).Should(BeNumerically(">", 0), "Job should fail")
+		ExpectWithOffset(2, err).ToNot(HaveOccurred())
+		for _, c := range job.Status.Conditions {
+			switch c.Type {
+			case batchv1.JobFailed:
+				if c.Status == k8sv1.ConditionTrue {
+					return true
+				}
+			case batchv1.JobComplete:
+				ExpectWithOffset(2, c.Status).NotTo(Equal(k8sv1.ConditionTrue), "Job should fail")
+			}
+		}
+		return false
+	}, timeoutSec*time.Second, 1*time.Second).Should(BeTrue(), "Job should fail")
 }
 
-func RenderJob(name string, cmd, args []string) *batchv1.Job {
+// NewJob creates a job configuration that runs a single Pod.
+// A name is used for the job & pod while the command and its arguments are passed to the pod for execution.
+// In addition, the following arguments control the job behavior:
+// retry: The number of times the job should try and run the pod.
+// ttlAfterFinished: The period of time between the job finishing and its auto-deletion.
+// timeout: The overall time at which the job is deleted, regardless of it finishing or not.
```

Comment:
I think that the job will not be deleted. It's just terminate.

Reply:
Right, I should have used "terminated".